pkg/router/router_test.go

```diff
@@ -157,24 +163,100 @@ func TestAdmissionEdgeCases(t *testing.T) {
 	}
 }
 
+func TestConfigTemplateExecution(t *testing.T) {
+	// watching for errors
+	caughtErrors := []error{}
+	errCh, stopCh := make(chan error), make(chan struct{})
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func(ch chan error, stop chan struct{}, wg *sync.WaitGroup) {
+		defer wg.Done()
+		for {
+			select {
+			case err := <-ch:
+				caughtErrors = append(caughtErrors, err)
+			case <-stop:
+				return
+			}
+		}
+	}(errCh, stopCh, wg)
+
+	// adding custom handler which pipes to the error channel
+	errHandlersBefore := utilruntime.ErrorHandlers
+	utilruntime.ErrorHandlers = append(utilruntime.ErrorHandlers, (&pipeErrorHandler{errCh}).handle)
+	defer func() { utilruntime.ErrorHandlers = errHandlersBefore }()
+
+	// create routes whose settings would add some additional blocks to the conf
+	start := time.Now()
+	tests := map[string][]expectation{
+		"long whitelist of IPs": {
+			mustCreate{
+				name: "w",
+				host: "anotherexample.com",
+				path: "",
+				time: start,
+				annotations: map[string]string{
+					"haproxy.router.openshift.io/ip_whitelist": getDummyIPs(100),
+				},
+				tlsTermination: routev1.TLSTerminationEdge,
+			},
+		},
+	}
+
+	for name, expectations := range tests {
+		for _, expectation := range expectations {
+			err := expectation.Apply(h)
+			if err != nil {
+				t.Fatalf("%s failed: %v", name, err)
+			}
+		}
+	}
+
+	// let the router reload
+	time.Sleep(reloadInterval * 2)
```

Comment:
Is this a synthetic sleep? Why does sleeping 200ms work here? Would it work without the multiplication?

Reply:
Yes, it's synthetic. I  wanted to be sure the reload happened, that's why waiting twice more that the reload interval.

As we discussed, I made a stress test of this test function, it looks stable:
```bash
$ go test -run=TestConfigTemplateExecution -count=1000 ./pkg/router/
ok  	github.com/openshift/router/pkg/router	201.083s
$ stress -p=4 ./router.test -test.run=TestConfigTemplateExecution
5s: 52 runs so far, 0 failures
10s: 108 runs so far, 0 failures
15s: 164 runs so far, 0 failures
20s: 224 runs so far, 0 failures
25s: 280 runs so far, 0 failures
30s: 336 runs so far, 0 failures
...
12m0s: 8122 runs so far, 0 failures
12m5s: 8178 runs so far, 0 failures
12m10s: 8234 runs so far, 0 failures
12m15s: 8290 runs so far, 0 failures
^C
```

Reply:
Thanks. 

/lgtm